File: src/系统/06．经济系统/00．宝箱系统/06．开启完成回调.lua
local ____lualib = require("lualib_bundle")
local __TS__ArraySlice = ____lualib.__TS__ArraySlice
local ____exports = {}
local callbacks = {}
local function _____5B89_5168_6267_884C_56DE_8C03(callback, unit, target, progressBar, openTime, chestConfig, ownerUnit)
    do
        pcall(function()
            callback(
                unit,
                target,
                progressBar,
                openTime,
                chestConfig,
                ownerUnit
            )
        end)
    end
end
____exports["注册宝箱开启完成回调"] = function(callback)
    if type(callback) ~= "function" then
        return
    end
    callbacks[#callbacks + 1] = callback
end
____exports["触发宝箱开启完成回调"] = function(unit, target, progressBar, openTime, chestConfig, ownerUnit)
    if #callbacks == 0 then
        return
    end
    local current = __TS__ArraySlice(callbacks)
    do
        local i = 0
        while i < #current do
            _____5B89_5168_6267_884C_56DE_8C03(
                current[i + 1],
                unit,
                target,
                progressBar,
                openTime,
                chestConfig,
                ownerUnit
            )
            i = i + 1
        end
    end
end
return ____exports

File: src/系统/06．经济系统/00．宝箱系统/06．开启完成回调.js
/** @noSelfInFile */
const { debugLogForce } = require("lib.扩展函数.自定义扩展函数.03．调试输出");
const 调试模块 = "宝箱完成回调";
const callbacks = [];
function 安全执行回调(callback, unit, target, progressBar, openTime, chestConfig, ownerUnit) {
    try {
        callback(unit, target, progressBar, openTime, chestConfig, ownerUnit);
    }
    catch (err) {
        debugLogForce(调试模块, "回调执行失败", "err=", err);
    }
}
export function 注册宝箱开启完成回调(callback) {
    if (typeof callback !== "function")
        return;
    callbacks.push(callback);
}
export function 触发宝箱开启完成回调(unit, target, progressBar, openTime, chestConfig, ownerUnit) {
    if (callbacks.length === 0)
        return;
    const current = callbacks.slice();
    for (let i = 0; i < current.length; i++) {
        安全执行回调(current[i], unit, target, progressBar, openTime, chestConfig, ownerUnit);
    }
}
